### PR TITLE
[yang] add set_owner to feature yang

### PR DESCRIPTION
**Pull request by arlakshm**
Signed-off-by: Arvindsrinivasan Lakshmi Narasimhan <arlakshm@microsoft.com>

#### Why I did it
Add the configuration for the set_owner in the `feature` yang model

#### How I did it
Add new leaf `set_pwner` to the `feature` yang model

#### How to verify it
compile `sonic_yang_mgmt-1.0-py3-none-any.whl`

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106

### CODE REVIEW

File: src/sonic-yang-models/yang-models/sonic-types.yang
Comment (arlakshm) [reply src/sonic-yang-models/yang-models/sonic-types.yang]: Fixed in the latest commit

**Comment by tw0byt3** [comment]
Hi Team,

When will this change be reviewed?

File: src/sonic-yang-models/tests/files/sample_config_db.json
Comment (arlakshm) [reply src/sonic-yang-models/tests/files/sample_config_db.json:1107]: We will take a look in future PR

**Comment by arlakshm** [comment]
@praveen-li, @dgsudharsan can you approve it there are not other comments.

**Comment by arlakshm** [comment]
@praveen-li, @dgsudharsan can you please approve it there are no other comments.

**Comment by tw0byt3** [comment]
@lguohan Sir, awaiting your review.

**Review by dgsudharsan — APPROVED** [review]

**Comment by tw0byt3** [comment]
@praveen-li can you please approve the changes

**Comment by tw0byt3** [comment]
@praveen-li Could you please review and approve the same? I have an ongoing project on hold due to this FEATURE yang module.

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/feature.json
Comment (renukamanavalan) [inline src/sonic-yang-models/tests/yang_model_tests/tests_config/feature.json:66]: Is this an intentional failure case?

Can there be a test case w/o "set_owner" entry ?

Comment (praveen-li) [reply src/sonic-yang-models/tests/yang_model_tests/tests_config/feature.json:66]: Also a test with wrong value for set_owner will be useful.

**Review by praveen-li — COMMENTED** [review]
Looks good overall, Plz add a negative test as well. Thanks.

**Comment by dgsudharsan** [comment]
Looks like there are some more changes needed in feature YANG.
Note: Below table(s) have no YANG models:
KDUMP, SNMP, SNMP_COMMUNITY, WJH, WJH_CHANNEL, XCVRD_LOG,
libyang[0]: Value "always_disabled" does not satisfy the constraint "enabled|disabled|always_enabled" (range, length, or pattern). (path: /sonic-feature:sonic-feature/FEATURE/FEATURE_LIST[name='mux']/state)
sonic_yang(3):Data Loading Failed:Value "always_disabled" does not satisfy the constraint "enabled|disabled|always_enabled" (range, length, or pattern).
Data Loading Failed
Value "always_disabled" does not satisfy the constraint "enabled|disabled|always_enabled" (range, length, or pattern).
ConfigMgmt Class creation failed
Failed to break out Port. Error: Failed to load the config. Error: ConfigMgmtDPB Class creation failed

The "always_disabled" is missing for the state but its generated from init_cfg.json. Can you please update in this PR?

**Review by dgsudharsan — CHANGES_REQUESTED** [review]
Please add always_disabled in state as requested

**Comment by arlakshm** [comment]
> Please add always_disabled in state as requested

Added 'always_disabled' to `feature-state` in the latest commit

File: src/sonic-yang-models/tests/yang_model_tests/tests_config/feature.json
Comment (arlakshm) [reply src/sonic-yang-models/tests/yang_model_tests/tests_config/feature.json]: add new test in the latest commit

**Review by renukamanavalan — DISMISSED** [review]

**Comment by renukamanavalan** [comment]
@praveen-li - Waiting for your approval.

**Review by praveen-li — APPROVED** [review]
LGTM

File: src/sonic-yang-models/yang-models/sonic-feature.yang
Comment (venkatmahalingam) [inline src/sonic-yang-models/yang-models/sonic-feature.yang:18]: If we could add some description on kube and local, it would be helpful for users.

**Review by dgsudharsan — APPROVED** [review]

File: src/sonic-yang-models/yang-models/sonic-feature.yang
Comment (arlakshm) [reply src/sonic-yang-models/yang-models/sonic-feature.yang:18]: @venkatmahalingam, the description is added `set_owner` leaf where `feature-owner` is used.

**Review by ganglyu — APPROVED** [review]
LGTM

File: src/sonic-yang-models/yang-models/sonic-feature.yang
Comment (venkatmahalingam) [inline src/sonic-yang-models/yang-models/sonic-feature.yang:13]: Please add short description of the states as well.

Comment (arlakshm) [reply src/sonic-yang-models/yang-models/sonic-feature.yang:14]: added in latest commit

**Review by dgsudharsan — APPROVED** [review]

**Comment by dgsudharsan** [comment]
@arlakshm Can you please sync to the latest to check if pipeline passes? I see pipeline passing in other few other PRs in sonic-buildimage

**Comment by zhangyanzhao** [comment]
build failure need be fixed.

**Comment by zhangyanzhao** [comment]
@qiluo-msft  will review and merge once build check passed

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**
